### PR TITLE
APP-1420 adding multi-select tests

### PR DESCRIPTION
**Pull request by RoxyFarhad**

### CODE REVIEW

File: tests/multi-select.spec.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
+import { test, expect } from '@playwright/test';
+import { waitForCustomEvent, waitForCustomEventWithParam } from './lib/helper.ts';
+
+test.beforeEach(async ({page}) => {
+    await page.goto('/multi-select-test.html')
+})
+
+test('Given a default multiselect, test options, labels and are visible and selecting, and clearing an options should correctly set pills', async ({page}) => {
```
Comment (ehhong) [inline tests/multi-select.spec.ts:8]: [nit] wording is a little confusing, maybe 
"Given a default multiselect, shows test options and labels, and sets pills correctly after selecting and clearing options"

Comment (ehhong) [reply tests/multi-select.spec.ts:8]: would add in the description that this test is also checking for correctly fired events

File: tests/multi-select.spec.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    await expect(optionsContainer.locator('label', {hasText: 'sad'}).first().locator('input')).toBeChecked()
+})
+
+test('Given a multi-select with button, there is a button at the bottom', async ({page}) => {
+    const multiselect = page.getByTestId('multiselect-with-button')
+    await expect(multiselect).toBeVisible()
+    await multiselect.click()
+    await expect(multiselect.locator('v-select-button').first()).toBeVisible()
+    await expect(multiselect.locator('v-select-button').first()).toHaveText(/TAKE PHOTO/)
+    await expect(multiselect.locator('v-select-button').first().locator('v-icon').first()).toBeVisible()
```
Comment (ehhong) [inline tests/multi-select.spec.ts:88]: add a check that it's the correct icon?

File: tests/multi-select.spec.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    
+    const buttonClickevent = waitForCustomEvent(page, 'button-click')
+    await multiselect.locator('v-select-button').first().click()
+    expect(buttonClickevent).toBeTruthy()
+
+})
+
+test('Given a multi-select with a header, there is a header in the optiosn list', async ({page}) => {
```
Comment (ehhong) [inline tests/multi-select.spec.ts:96]: [nit] typo, "optiosn"

File: tests/multi-select.spec.ts [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    await expect(multiselect.locator('.options-container').first()).toBeVisible()
+    await expect(multiselect.locator('.options-container').first().locator('button', {hasText: 'Clear all'})).not.toBeVisible()
+})
+
+test('Given a tooltip and state=info, the select should show an info icon with a tooltip', async ({page}) => {
+    const multiselect = page.getByTestId('multiselect-info-tooltip')
+    await expect(multiselect).toBeVisible()
+    await expect(multiselect.locator('v-tooltip').first()).toBeVisible()
+    await expect(multiselect.locator('v-tooltip').first().locator('div').first()).toHaveClass(/icon-info-outline/)
+    await multiselect.locator('v-tooltip').first().hover()
```
Comment (ehhong) [inline tests/multi-select.spec.ts:135]: don't think you need to test the tooltip's functionality (should be covered in tooltip tests) but doesn't hurt much either

File: tests/multi-select.spec.ts [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+test('Given a tooltip and state=error the select should show a error icon with tooltip', async ({page}) => {
+    const multiselect = page.getByTestId('multiselect-error-tooltip')
+    await expect(multiselect).toBeVisible()
+    await expect(multiselect.locator('v-tooltip').first()).toBeVisible()
+    await expect(multiselect.locator('v-tooltip').first().locator('div').first()).toHaveClass(/text-danger-fg/)
+    await multiselect.locator('v-tooltip').first().hover()
+    await expect(multiselect.getByText(/error tip/)).toBeVisible()
+})
+
+test('Setting showpill=false should stop pils being rendered', async ({page}) => {
```
Comment (ehhong) [inline tests/multi-select.spec.ts:157]: [nit] typo "pils"

**Review by ehhong — APPROVED** [review]
lgtm, just some small nits!

File: tests/multi-select.spec.ts [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+    await expect(multiselect.getByText(/info tip/)).toBeVisible()
+})
+
+test('Given a tooltip and state=warn the select should show a warn icon with tooltip', async ({page}) => {
+    const multiselect = page.getByTestId('multiselect-warn-tooltip')
+    await expect(multiselect).toBeVisible()
+    await expect(multiselect.locator('v-tooltip').first()).toBeVisible()
+    await expect(multiselect.locator('v-tooltip').first().locator('div').first()).toHaveClass(/text-warning-fg/)
+    await multiselect.locator('v-tooltip').first().hover()
+    await expect(multiselect.gedtByText(/warn tip/)).toBeVisible()
```
Comment (anjinai) [inline tests/multi-select.spec.ts:145]: [nit] typo - should be getByText

**Pull request merged**